backend/src/map_model.rs

```diff
@@ -158,6 +158,9 @@ impl Intersection {
             if to_road.id == from_road.id {
                 return None;
             }
+            if router_input.has_modal_filter(to_road.id) {
+                return None;
```

Comment:
I think we will relax this later to solve part of #181. When a route starts or ends on a filtered road, we can travel only one direction along it, depending on the start position.